tg_bot.py

```diff
@@ -14,26 +16,28 @@
                     level=logging.INFO)
 logger = logging.getLogger(__name__)
 
-"""
-def nawab_twitter_authenticate():
-    auth = tweepy.OAuthHandler(config.consumer_key, config.consumer_secret)
-    auth.set_access_token(config.access_token_key, config.access_token_secret)
-    api = tweepy.API(auth)
-    return api
-"""
+auth = tweepy.OAuthHandler(config.consumer_key, config.consumer_secret)
+auth.set_access_token(config.access_token_key, config.access_token_secret)
+api = tweepy.API(auth)
 
+default_dir = '/var/log/nawab/'
+
+u_id = pwd.getpwuid( os.getuid() ).pw_name
+
+if not os.path.exists(default_dir):
+    os.system(("sudo mkdir %s" % (default_dir)))
+
+ownership_command = "sudo chown %s: %s" % (u_id, default_dir)
+os.system(ownership_command)
```

Comment:
This still adds more redundancy. You are just copy-pasting code from twitter bot to here. Rather you should try to figure out a way where you can import things.

Or otherwise, look into the recent PR I just opened on integrating both bots into one interface. That should help you sort this out.